From d8804ced2a3b0ef4ff79b02e80f153e546e54d6b Mon Sep 17 00:00:00 2001
From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Mon, 1 Sep 2014 09:25:06 -0400
Subject: [md] don't call md_allow_write in get_bitmap_file

Message-id: <1409563584-9000-5-git-send-email-Jes.Sorensen@redhat.com>
Patchwork-id: 89937
O-Subject: [RHEL7.1 PATCH 04/82] md: don't call md_allow_write in get_bitmap_file.
Bugzilla: 1085530
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

From: NeilBrown <neilb@suse.de>

There is no really need as GFP_NOIO is very likely sufficient,
and failure is not catastrophic.

Calling md_allow_write here will convert a read-auto array to
read/write which could be confusing when you are just performing
a read operation.

Signed-off-by: NeilBrown <neilb@suse.de>
(cherry picked from commit 60559da4d8c3259ea41a14ca4cfcd83022bff6bd)
Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/md/md.c b/drivers/md/md.c
index 9430555..8406901 100644
--- a/drivers/md/md.c
+++ b/drivers/md/md.c
@@ -5647,10 +5647,7 @@ static int get_bitmap_file(struct mddev * mddev, void __user * arg)
  char *ptr, *buf = NULL;
  int err = -ENOMEM;
 
- if (md_allow_write(mddev))
-  file = kmalloc(sizeof(*file), GFP_NOIO);
- else
-  file = kmalloc(sizeof(*file), GFP_KERNEL);
+ file = kmalloc(sizeof(*file), GFP_NOIO);
 
  if (!file)
   goto out;
-- 
1.7.1